matching-core/src/main/java/com/graphhopper/matching/MapMatching.java

```diff
@@ -301,70 +328,92 @@ private void computeTransitionProbabilities(TimeStep<GPXExtension, GPXEntry, Pat
         for (GPXExtension from : prevTimeStep.candidates) {
             for (GPXExtension to : timeStep.candidates) {
                 RoutingAlgorithm algo = algoFactory.createAlgo(queryGraph, algoOptions);
-                // System.out.println("algo " + algo.getName());
                 final Path path = algo.calcPath(from.getQueryResult().getClosestNode(),
                         to.getQueryResult().getClosestNode());
                 if (path.isFound()) {
                     timeStep.addRoadPath(from, to, path);
                     final double transitionLogProbability = probabilities
                             .transitionLogProbability(path.getDistance(), linearDistance, timeDiff);
                     timeStep.addTransitionLogProbability(from, to, transitionLogProbability);
+                } else {
+                    // TODO: can we remove maxVisitedNodes completely and just set to infinity?
+                    if (algo.getVisitedNodes() > algoOptions.getMaxVisitedNodes()) {
+                        throw new RuntimeException(
+                                "couldn't compute transition probabilities as routing failed due to too small maxVisitedNodes ("
+                                        + algoOptions.getMaxVisitedNodes() + ")");
+                    }
+                    // TODO: can we somewhere record that this route failed? Currently all viterbi
+                    // knows is that there's no transition possible, but not why. This is useful
+                    // for e.g. explaining why a sequence broke.
```

Comment:
If we split into multiple MatchResult (one per each sequence) then maybe we could add a 'sequenceStartReason' property? For the first sequence it will be 'first sequence' and for the rest it might be 'no candidates found' or 'no transitions found' etc. @karussell / @stefanholder ?